src/wallet/rpcwallet.cpp

```diff
@@ -3297,16 +3297,18 @@ UniValue rescanblockchain(const JSONRPCRequest& request)
         }
     }
 
-    const CBlockIndex* stopBlock = pwallet->ScanForWalletTransactions(pindexStart, pindexStop, reserver, true);
-    if (!stopBlock) {
-        if (pwallet->IsAbortingRescan()) {
-            throw JSONRPCError(RPC_MISC_ERROR, "Rescan aborted.");
-        }
-        // if we got a nullptr returned, ScanForWalletTransactions did rescan up to the requested stopindex
+    const CBlockIndex* stopBlock = nullptr;
```

Comment:
nit: No need to initialize a return value. In fact this might look like you are passing in something.